target/ppc/int_helper.c

```diff
@@ -1706,6 +1706,41 @@ VEXTRACT(uw, u32)
 VEXTRACT(d, u64)
 #undef VEXTRACT
 
+#define VSTRI(NAME, ELEM, SIZE, LEFT) \
+void helper_##NAME(CPUPPCState *env, ppc_avr_t *t, ppc_avr_t *b,    \
+                   target_ulong rc)                                 \
+{                                                                   \
+    const int array_size = ARRAY_SIZE(b->u##SIZE);                  \
```

Comment:
This is the only use of `SIZE`, so why not just use it directly in the loop and `idx` instead of `array_size` (and adjust the values)?

Reply:
Something like:
```c
#define VSTRI(NAME, ELEM, SIZE, LEFT) \
void helper_##NAME(CPUPPCState *env, ppc_avr_t *t, ppc_avr_t *b,    \
                   target_ulong rc)                                 \
{                                                                   \
    bool null_found = false;                                        \
    int i, idx;                                                     \
                                                                    \
    for (i = 0; i < SIZE; i++) {                                    \
        idx = LEFT ? i : SIZE - i - 1;                              \
        if (b->Vsr##ELEM(idx)) {                                    \
            t->Vsr##ELEM(idx) = b->Vsr##ELEM(idx);                  \
        } else {                                                    \
            null_found = true;                                      \
            break;                                                  \
        }                                                           \
    }                                                               \
                                                                    \
    for (; i < SIZE; i++) {                                         \
        idx = LEFT ? i : SIZE - i - 1;                              \
        t->Vsr##ELEM(idx) = 0;                                      \
    }                                                               \
                                                                    \
    if (rc && null_found) {                                         \
        env->crf[6] = 0b0010;                                       \
    } else {                                                        \
        env->crf[6] = 0;                                            \
    }                                                               \
}
VSTRI(VSTRIBL, B, 16, true)
VSTRI(VSTRIBR, B, 16, false)
VSTRI(VSTRIHL, H, 8, true)
VSTRI(VSTRIHR, H, 8, false)
#undef VSTRI
```
? I think it can be done. The idea was to make it clearer where the value comes from, but this way can be bit more direct.

Reply:
Yeah like that, but if you want to leave it with `array_size` it'd still look good to me

Reply:
Fixed in https://github.com/PPC64/qemu/pull/47/commits/a3758d0cddfc20e12fe4f8761b3a746c6ce4a3d6